### PR TITLE
Suggested namespace rename 

### PR DESCRIPTION
**Pull request by dsyme**
I suppose this is a form of bike-shedding, but the source in our tree doesn't match our binary names, which irks me every time I come to it.  We refer to
* FSharp.Compiler.Private.dll
* FSharp.Compiler.Service.dll
* FSharp.Compiler.Interactive.Settings.dll

and so on but our source is always `Microsoft.FSharp.Compiler`. 

Among other things this means you see a wall of text whenever you look at the `open` declarations at the head of the file on the F# compiler. Personally I think the compiler code looks simpler and more approachable with the shorter, plainer names - everyone knows Microsoft contribute heavily to F# and package it, we don't need to tell them 1700 times in the compiler code.

So I suggest we rename `Microsoft.FSharp.Compiler` --> `FSharp.Compiler` in our _source_.

* This only affects the internal compiler code and the API of FSharp.Compiler.Service.dll

* The one exception is `FSharp.Compiler.Interactive.Settings`, where `open Microsoft.FSharp.Compiler.Interactive.Settings` can be opened from existing F# scripts.  I've added 10 lines to make sure still works

* This doesn't affect `FSharp.Core`, which must remain binary compatible. (As background: Back in F# 3.1 we did a while lot of work to allow "open FSharp.Core" to be used instead of "open Microsoft.FSharp.Core".  This is in line with that but we can simply rename the source)

* This doesn't affect the name of the `Microsoft.FSharp.Compiler` nuget package which is built as part of Microsoft's the distribution of F# in the .NET SDK.  

* This doesn't affect the Visual F# bits.

* This also does `Microsoft.FSharp.Build` --> `FSharp.Build` - again the DLL name is not changed, and the `UsingTasks` references don't refer to namespaces

### CODE REVIEW

**Comment by 7sharp9** [comment]
Ooooh, this would be a good time to check for internal reflection calls :-)

The namespace discrepancy is quite annoying.

File: fcs/RELEASE_NOTES.md [hunk 1]
```diff
@@ -458,7 +458,7 @@
 * Replace GetSymbolAtLocationAlternate by GetSymbolUseAtLocation
 
 #### 0.0.40 - 
-* Fixed #86 - Expose Microsoft.FSharp.Compiler.Interactive.Shell.Settings.fsi
+* Fixed #86 - Expose FSharp.Compiler.Interactive.Shell.Settings.fsi
```
Comment (auduchinok) [inline fcs/RELEASE_NOTES.md]: Should it really be changed here?

**Comment by cartermp** [comment]
This is a good change to make, assuming there aren't some policy issues that come up.

**Comment by dsyme** [comment]
@cartermp Same test failing again, will need to look into this:

```
   at FSharp.Compiler.Service.Tests.PerfTests.Test request for parse and check doesn't check whole project() in D:\a\1\s\tests\service\PerfTests.fs:line 75
```

**Comment by dsyme** [comment]
Another failure in `Test request for parse and check doesn't check whole project`

Will re-run and put in a separate PR to diagnose that test

**Pull request merged**